src/app/home/components/DetailViewItem.tsx

```diff
@@ -51,61 +61,95 @@ const DetailViewItem: React.FunctionComponent<IProps> = ({
   onAddItem,
   type,
   onSearch,
+  onStageTriggered,
+  onMigrateTriggered,
+  associatedPlans,
+  onWizardToggle,
+  isLoading,
   ...props
 }) => {
+  let listComponent;
+  if (type === 'cluster') {
+    listComponent = (
+      <ClusterDataListComponent
+        onRemoveItem={onRemoveItem}
+        dataList={filteredDataList}
+        associatedPlans={associatedPlans}
+      />
+    );
+  } else if (type === 'storage') {
+    listComponent = (
+      <StorageDataListComponent
+        onRemoveItem={onRemoveItem}
+        dataList={filteredDataList}
+        associatedPlans={associatedPlans}
+      />
+    );
+  } else {
+    listComponent = (
+      <PlanDataListComponent
+        onRemoveItem={onRemoveItem}
+        onStageTriggered={onStageTriggered}
+        onMigrateTriggered={onMigrateTriggered}
+        dataList={filteredDataList}
+        onWizardToggle={onWizardToggle}
+        isLoading={isLoading}
+        plansDisabled={plansDisabled}
+      />
+    );
+  }
+
   return (
-    <Card>
-      <CardBody>
-        <DataListItem aria-labelledby="ex-item1" isExpanded={isExpanded}>
-          <Flex width="100%" height="5em" >
-            <Box flex="0 0 2em" my="auto">
-              <DataListToggle
-                onClick={() => onToggle(id)}
-                isExpanded={isExpanded}
-                id={id}
-              />
-            </Box>
-            <Box flex="1" my="auto">
-              {title}
-            </Box>
-            <Box textAlign="left" flex="0 0 15em" my="auto">
-              {addButton}
-            </Box>
-          </Flex>
-          <DataListContent
-            aria-label="Primary Content Details"
-            isHidden={!isExpanded}
-          >
-            {(allData && allData.length) > 0 &&
-              <InputGroup
-                // @ts-ignore
-                css={css`
+    <DataListItem aria-labelledby="ex-item1" isExpanded={isExpanded}>
```

Comment:
Rename DetailViewItem & DataListItem. Refactor this code to use composition since it is reusing. Need to decide if we are using HOC/ render props for this.